src/components/Header.tsx

```diff
@@ -0,0 +1,110 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { Errors } from '../types/Errors';
+
+interface Props {
+  todos: Todo[],
+  title: string,
+  userId: number,
+  isDisabledInput: boolean,
+  setTodos: (todos: Todo[]) => void,
+  setTitle: (str: string) => void,
+  handleUpdateTodo: (todo: Todo) => void,
+  handleAddTodo: (todo: Omit<Todo, 'id'>) => void
+  setErrorMessage: (message: Errors | '') => void
+}
+
+export const Header: React.FC<Props> = ({
+  todos,
+  title,
+  userId,
+  isDisabledInput,
+  setTitle,
+  setTodos,
+  handleAddTodo,
+  setErrorMessage,
+  handleUpdateTodo,
+}) => {
+  const [isCheckedAll, setIsCheckedAll] = useState(false);
+  const inputField = useRef<HTMLInputElement>(null);
+
+  const allTodosChecked = todos.every(todo => todo.completed);
+
+  useEffect(() => {
+    const newTodos = [...todos]
+      .map(todo => ({ ...todo, completed: isCheckedAll }));
```

Comment:
You are creating a new array from 'todos' but it's not necessary here because .map() already returns a new array.